### ISSUE TITLE
WebGPU shading language

### ISSUE THREAD

**Issue by Korijn** [almarklein/wgpu-py#66]
https://gpuweb.github.io/gpuweb/wgsl.html

Well, there it is! Do we need to do anything in response?

**Comment by almarklein** [comment]
Nice! Mmm, it's very early days, but we should keep a close eye, and on the tooling. Would be nice if we can eventually support it (via python-shader?), or perhaps even adopt it.

**Comment by Korijn** [comment]
I suspect there will be a generally available compiler (wgsl to spirv) so we could create a new lib to wrap it with python, or that may not even be necessary if it has a decent API and we can just let the user provide their spirv bytes, I think?

**Comment by almarklein** [comment]
Yeah, users should be able to bring their own SpirV files anyway, so technically we already support it. We could make an example to demonstrate it (here, or in our higher level libs).

**Comment by Korijn** [comment]
> Yeah, users should be able to bring their own SpirV files anyway, so technically we already support it. We could make an example to demonstrate it (here, or in our higher level libs).

Awesome!

**Comment by almarklein** [comment]
Moving this issue to the python-shader repo. Which is probably where we should do something, if anything. In `wgpu-py` we don't deal with shaders.
